CMakeLists.txt

```diff
@@ -927,9 +907,8 @@ get_property(CSTD GLOBAL PROPERTY CSTD)
 set_ifndef(CSTD c99)
 
 # @Intent: Obtain compiler specific flag for specifying the c standard
-toolchain_cc_cstd_flag(CC_CSTD ${CSTD})
 zephyr_compile_options(
-  $<$<COMPILE_LANGUAGE:C>:${CC_CSTD}>
+  $<$<COMPILE_LANGUAGE:C>:$<TARGET_PROPERTY:compiler,cstd>${CSTD}>
```

Comment:
For compilers that do not support -std and hence the default c99, the global property CSTD can simply be set to empty string as the following:
`set_property(GLOBAL PROPERTY CSTD " ")`
and the cstd can then be left to use the template setting.

Reply:
I assume this is acceptable ?
Current Zephyr would require you to do:
```
 set_property(GLOBAL PROPERTY CSTD " ")
```
right ?

Reply:
This is perfectly acceptable. 
I only mentioned it as a hint that it might be needed by some compilers and that it was easily achievable by simply setting the global property to an empty string.